go/viperutil/internal/sync/sync_test.go

```diff
@@ -93,7 +93,10 @@ func TestWatchConfig(t *testing.T) {
 
 	sv := vipersync.New()
 	A := viperutil.Configure("a", viperutil.Options[int]{Dynamic: true})
-	B := viperutil.Configure("b", viperutil.Options[int]{Dynamic: true})
+	B := viperutil.Configure("b", viperutil.Options[int]{FlagName: "b", Dynamic: true, Default: 5})
+
+	// Check that default values are actually used
+	require.Equal(t, B.Get(), B.Default())
```

Comment:
Confirmed that this fails on main:
```
$ go test -timeout 30s -run ^TestWatchConfig$ vitess.io/vitess/go/viperutil/internal/sync

--- FAIL: TestWatchConfig (0.00s)
    /Users/matt/git/vitess/go/viperutil/internal/sync/sync_test.go:99: 
        	Error Trace:	/Users/matt/git/vitess/go/viperutil/internal/sync/sync_test.go:99
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 5
        	Test:       	TestWatchConfig
```

Reply:
Yes it did. I started with the test (created by @aquarapid) and went from there.